libvirt/tests/src/virtual_disks/virtual_disks_nbd.py

```
@@ -0,0 +1,282 @@
+
+        :param: save_file: the path to saved file
+        """
+        # Save the domain.
+        ret = virsh.save(vm_name, save_file,
+                         **virsh_dargs)
+        libvirt.check_exit_status(ret)
+
+        # Restore the domain.
+        ret = virsh.restore(save_file, **virsh_dargs)
+        libvirt.check_exit_status(ret)
+
+    def check_snapshot():
+        """
+        Check domain snapshot operations.
+        """
+        # Cleaup dirty data if exists
+        if os.path.exists(snapshot_name1_file):
+            os.remove(snapshot_name1_file)
+        if os.path.exists(snapshot_name2_mem_file):
+            os.remove(snapshot_name2_mem_file)
+        if os.path.exists(snapshot_name2_disk_file):
+            os.remove(snapshot_name2_disk_file)
+        device_target = 'vda'
+        snapshot_name1_option = "--diskspec %s,file=%s,snapshot=external --disk-only --atomic" % (device_target, snapshot_name1_file)
+        ret = virsh.snapshot_create_as(vm_name, "%s %s" % (snapshot_name1, snapshot_name1_option), debug=True)
+        libvirt.check_exit_status(ret)
+        snap_lists = virsh.snapshot_list(vm_name, debug=True)
+        if snapshot_name1 not in snap_lists:
+            test.fail("Snapshot %s doesn't exist"
+                      % snapshot_name1)
+        # Check file can be created after snapshot
+
+        def _check_file_create(filename):
+            try:
+                session = vm.wait_for_login()
+                if platform.platform().count('ppc64'):
+                    time.sleep(10)
+                cmd = ("echo"
+                       " teststring > /tmp/{0}".format(filename))
+                status, output = session.cmd_status_output(cmd)
+                if status != 0:
+                    test.fail("Failed to touch one file on VM internal")
+            except (remote.LoginError, virt_vm.VMError, aexpect.ShellError) as e:
+                logging.error(str(e))
+                raise
```

If only logging the error, why not just remove the except and let them raise directly?

---

logging the error is more human readable for troubleshooting